features/steps/validation_handling.py

```diff
@@ -0,0 +1,281 @@
+
+                        validation_outcome = ValidationOutcome(
+                            outcome_code=getattr(ValidationOutcomeCode, validation_outcome.outcome_code),
+                            # observed = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            # expected = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            observed=validation_outcome.model_dump(include=('observed'))["observed"],  # TODO (parse it correctly)
+                            expected=validation_outcome.model_dump(include=('expected'))["expected"],  # TODO (parse it correctly)
+                            feature=context.feature.name,
+                            feature_version=misc.define_feature_version(context),
+                            severity=getattr(OutcomeSeverity, "WARNING" if any(tag.lower() == "warning" for tag in context.feature.tags) else "ERROR"),
+                            check_execution_id=check_execution_id
+                        )
+                        context.gherkin_outcomes.append(validation_outcome)
+
+                    if not step_results:
+
+                        if isinstance(inst, (tuple, list)): # TODO -> this is quite dirty temp solution. Done because @given('Its attribute {attribute}') return a tuple, not an instance
+                            inst = inst[0]
+
+                        StepOutcome(inst=activation_inst,
+                                    context=context,
+                                    expected=None,
+                                    observed=None)  # expected / observed equal on passed rule?
+
+                        validation_outcome = ValidationOutcome(
+                            outcome_code=ValidationOutcomeCode.P00010,  # "Rule passed"
+                            # observed = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            # expected = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            observed='',  # TODO (parse it correctly)
+                            expected='',  # TODO (parse it correctly)
+                            feature=context.feature.name,  #
+                            feature_version=misc.define_feature_version(context),
+                            severity=OutcomeSeverity.PASS,
+                            check_execution_id=check_execution_id
+                        )
+                    context.gherkin_outcomes.append(validation_outcome)
+            
+                generate_error_message(context, [gherkin_outcome for gherkin_outcome in context.gherkin_outcomes if gherkin_outcome.severity == OutcomeSeverity.ERROR])
```

Comment:
```suggestion
                generate_error_message(context, [gherkin_outcome for gherkin_outcome in context.gherkin_outcomes if gherkin_outcome.severity >= OutcomeSeverity.WARNING])
```

Probably do warnings as well. That's the nice thing about a consistent order in the severities that we can use > here.

Reply:
Good point. Adding.